Sources/Controllers/2.0/RoutePlanning/OASelectTrackFolderViewController.m

```diff
@@ -0,0 +1,260 @@
+//
+//  OASelectTrackFolderViewController.m
+//  OsmAnd
+//
+//  Created by nnngrach on 05.02.2021.
+//  Copyright © 2021 OsmAnd. All rights reserved.
+//
+
+#import "OASelectTrackFolderViewController.h"
+#import "OAColors.h"
+#import "Localization.h"
+#import "OAUtilities.h"
+#import "OASettingsTableViewCell.h"
+#import "OATitleRightIconCell.h"
+#import "OAMultiIconTextDescCell.h"
+#import "OAAddTrackFolderViewController.h"
+#import "OsmAndApp.h"
+#import "OALoadGpxTask.h"
+
+#define kCellTypeAction @"OATitleRightIconCell"
+#define kMultiIconTextDescCell @"OAMultiIconTextDescCell"
+#define kAddNewFolderSection 0
+#define kFoldersListSection 1
+
+@interface OASelectTrackFolderViewController() <UITableViewDelegate, UITableViewDataSource, OAAddTrackFolderDelegate>
+
+@end
+
+@implementation OASelectTrackFolderViewController
+{
+    id<OASelectTrackFolderDelegate> _delegate;
+    OAGPX *_gpx;
+    NSArray<NSArray<NSDictionary *> *> *_data;
+}
+
+- (instancetype) initWithGPX:(OAGPX *)gpx delegate:(id<OASelectTrackFolderDelegate>)delegate;
```

Comment:
As a rule, you do not pass the delegate via the constructor. Instead, you should use public weak property and set it after an instance is created.